SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduledJobPoller.java

```diff
@@ -112,10 +112,20 @@ public void runActionOnPoll() {
 
       final CronExpression cronExpression;
 
+      Optional<String> maybeSchedule = request.getRequest().getQuartzScheduleOrSchedule();
+      if (!maybeSchedule.isPresent()) {
```

Comment:
That shouldn't be possible in this spot, but we just encountered (and, to be fair, fixed) a bug where the validator was relying on the method before checking if either schedule was there. So maybe that should raise an exception if the schedule isn't present and log an error (rather than a warning), because the only way it'll be the case is if something went wrong.